src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,72 @@
+package duke.util;
+
+import java.util.Objects;
+
+import duke.command.AddCommand;
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Parses commands and checks it.
+ * @author Jicson Toh
+ */
+public class Parser {
+
+    /**
+     * Checks the input command.
+     * @param action user input.
+     * @return returns the string action.
+     */
+    public Command parseCommand(String action) throws DukeException {
```

Comment:
I like that it systematically checks for the single word commands then for the other commands 👍

Reply:
Thanks!